streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -87,10 +87,13 @@ public void init(final InternalProcessorContext context) {
             this.valDeserializer = (Deserializer<V>) context.valueSerde().deserializer();
         }
 
-        // if value deserializers are internal wrapping deserializers that may need to be given the default
+        // if deserializers are internal wrapping deserializers that may need to be given the default
         // then pass it the default one from the context
         if (valDeserializer instanceof WrappingNullableDeserializer) {
-            ((WrappingNullableDeserializer) valDeserializer).setIfUnset(context.valueSerde().deserializer());
+            ((WrappingNullableDeserializer) valDeserializer).setIfUnset(
```

Comment:
@guozhangwang This is 2/2 of the areas that needed the fix. Same thing, the valueSerde was being passed into the underlying Serde, despite it really needing the keySerde.

Reply:
Ack, thanks!